### PR TITLE
RHIROS-510 add state value in details page

### PR DESCRIPTION
**Pull request by r14chandra**
Screenshot from 2022-02-16 15-31-54

### CODE REVIEW

File: src/Routes/RosSystemDetail/ros-details-page.scss [hunk 1]
```diff
@@ -12,4 +12,7 @@
       }
     }
   }
+}
+.state-value-gap {
+  margin-right: 107px;
```
Comment (PreetiW) [inline src/Routes/RosSystemDetail/ros-details-page.scss:17]: @r14chandra can you please tell me how did you come up with this value for right margin?

**Review by PreetiW — COMMENTED** [review]
@r14chandra Thank you for creating the PR, overall looks good. I have put a small query around the styling.

File: src/Routes/RosSystemDetail/ros-details-page.scss (hunk 1 above)
Comment (r14chandra) [reply src/Routes/RosSystemDetail/ros-details-page.scss:17]: Just to make it consistent with the gap between `Provider`, `Instance type` and others key-value pairs, I tried to match it. It's achievable at `107px` for `State` key by setting it for margin-right property . Do you have any other suggestions?

File: src/Routes/RosSystemDetail/RosSystemDetail.js [hunk 2]
```diff
@@ -36,11 +37,16 @@ class RosSystemDetail extends React.Component {
...
+                rating,
+                state: state
             } = this.props.rosSystemInfo;
             const { inventoryId } = this.props.match.params;
             return (
                 <Grid className='ros-system-info'>
+                    <GridItem>
+                        <span className='state-value-gap'>State:</span>
+                        <SystemState stateValue={ state }/>
+                    </GridItem>
```
Comment (PreetiW) [inline src/Routes/RosSystemDetail/RosSystemDetail.js:49]: @r14chandra after spending sometime going trhough patternfly component my suggestion would be to replace this block by using [`DescriptionList`](https://www.patternfly.org/v4/components/description-list/#compact-horizontal)  and refactor the `Expanded Row` component to use the same. This way we can keep it consistent and the styling will be handled by patternfly. I will also put the styling suggestion which you will need to change.

```suggestion
                  <DescriptionList className='expanded-row' isCompact isHorizontal>
                            <DescriptionListGroup>
                                <DescriptionListTerm>State</DescriptionListTerm>
                                 <DescriptionListDescription>
                                       <SystemState stateValue={ state }/>
                                  </DescriptionListDescription>
                            </DescriptionListGroup>
                        </DescriptionList>
```

for reference go through the conversation [here](https://patternfly.slack.com/archives/C4FM977N0/p1645451871080709) and [sandbox example here](https://codesandbox.io/s/bold-violet-426xfn?file=/index.js:684-704)

feel free to reach out to me for any queries!

File: src/Routes/RosSystemDetail/ros-details-page.scss [hunk 3]
```diff
@@ -12,4 +12,7 @@
       }
     }
   }
+}
```
Comment (PreetiW) [inline src/Routes/RosSystemDetail/ros-details-page.scss:15]: @r14chandra here is the styles you can use with the `DescriptionList` 

```suggestion
.ros-system-info {
  .pf-c-description-list.expanded-row {    
    dt {
      font-weight: var(--pf-global--FontWeight--normal);
      // width: 120px;
      &:after {
        content: ":";
      }
    }
  }

  .pf-c-description-list__group {
    row-gap: 0;
  }
}
```

File: src/Routes/RosSystemDetail/ros-details-page.scss (hunk 1 above)
Comment (PreetiW) [reply src/Routes/RosSystemDetail/ros-details-page.scss:17]: @r14chandra I have put the suggestion in the comments above, please check and let me know what you think!

**Review by PreetiW — CHANGES_REQUESTED** [review]
@r14chandra I have put the suggestion which will help us to be more consistent with the styles, let me know what you think.

File: src/Routes/RosSystemDetail/ros-details-page.scss (hunk 1 above)
Comment (PreetiW) [reply src/Routes/RosSystemDetail/ros-details-page.scss:17]: attaching the screenshot with the `DescriptionList` changes for your reference

image

File: src/Routes/RosSystemDetail/RosSystemDetail.js (hunk 2 above)
Comment (r14chandra) [reply src/Routes/RosSystemDetail/RosSystemDetail.js]: Thanks @PreetiW for digging deep 🚀 . This looks great to me! I've added the changes as suggested.

File: src/Routes/RosSystemDetail/RosSystemDetail.js [hunk 4]
```diff
@@ -36,11 +43,22 @@ class RosSystemDetail extends React.Component {
                 cloud_provider: cloudProvider,
                 instance_type: instanceType,
                 idling_time: idlingTime,
-                rating
+                rating,
+                state: state
```
Comment (PreetiW) [inline src/Routes/RosSystemDetail/RosSystemDetail.js:47]: @r14chandra FYI Suggeston: if we directly want to use the object key without changing the name then we can write just `state` here ( just like we did for rating). You can read more about  it here [Rename & Destructure Variables in ES6](https://wesbos.com/destructuring-renaming)

**Review by PreetiW — APPROVED** [review]
@r14chandra Thank you for making the changes based on the suggestion 💯 LGTM 🎉

**Pull request merged**